src/commands/compile/compileSingleFile.ts

```diff
@@ -53,31 +46,20 @@ export async function compileSingleFile(
     subCommand = await identifySasFile(target, sourcePath)
   }
 
-  let sourcefilePathParts = sourcePath.split(path.sep)
```

Comment:
All this logic has been extracted into `CompileComand`, since source path is a prerequisite for compilation.